### ISSUE TITLE
A770-RESIDENCY-001: move support operations and state to A770 and prove full-device-resident inference

### ISSUE THREAD

**Issue by EffortlessSteven** [EffortlessMetrics/bitnet-rs-swarm#1908]
## Parent

#1894

## Depends on

The trusted hybrid appliance must be healthy first:

- #1896 — correctness/parity;
- #1897–#1903 — resident QK256 runtime and complete execution ownership;
- #1904 and #1906 — quality/reliability and benchmark qualification;
- #1905 — shared operator/resident-session UX.

This issue must not block delivery or support promotion of the trusted hybrid route.

## Problem

Resident QK256 linears are not full-device-resident inference. The current model path still has CPU-owned embeddings, norms, RoPE, attention, KV state, residuals, final norm, tied output head/logits, and sampling boundaries. Moving these operations changes data layouts, state ownership, transfer boundaries, numerical behavior, memory pressure, and recovery semantics.

A full-residency claim requires operation-by-operation evidence, not inference from the selected backend label.

## Goal

Build and prove an exact official-model A770 route in which all required model operations and persistent inference state remain on the selected A770 across prefill and autoregressive decode, except explicitly permitted final control-plane/sampling work.

## Required subpaths

At minimum:

```text
embedding lookup
RMSNorm variants
residual adds
RoPE
Q/K/V/O and FFN linears
KV cache allocate/read/write/append
attention score computation
causal masking
softmax
value mix
FFN activation / elementwise multiply
final norm
tied LM head / logits
bounded top-k or logits handoff
```

Sampling may remain on the CPU initially only if the contract defines the minimal logits/top-k transfer and does not imply sampler residency.

## Architecture rules

- Use the execution-plan vocabulary from #1903.
- Keep operator kernels format/provider-specific; do not force all operations into one mega-backend trait.
- Introduce accelerator-resource abstractions only where at least two real operations need the same context/buffer/queue/event contract.
- Do not make the existing `bitnet-gpu-hal` monolith a production dependency merely to obtain a residency label; adopt reusable concepts only through reviewed extraction/adapters.
- No approximate or replacement attention rule may be substituted for the official model semantics.

### PR 1 — resident tensor/state contract

- Typed device tensor/buffer identity, shape/dtype/layout, context generation, ownership, lifetime, transfer, and receipt fragments.
- Characterize current CPU islands and transfers.
- No operation migration.

### PR 2 — embedding residency

- Exact tied embedding source and lookup semantics.
- CPU/A770 parity and generated-token preservation.

### PRs 3–5 — norms, residuals, and RoPE

- One semantic family per PR.
- Candidate-off/on parity and explicit chaining with resident inputs/outputs.

### PRs 6–9 — attention and KV

- KV layout/lifecycle first.
- Score/mask, softmax, and value mix as separately provable slices.
- Short and long decode parity; cancellation/reset cannot leave stale KV.

### PRs 10–11 — FFN elementwise and complete resident block

- ReLU²/activation and elementwise products.
- Complete layer/block resident chaining with no host round-trip.

### PRs 12–13 — final norm and tied LM head

- Exact output-head orientation/tied embedding semantics.
- Complete logits comparison and bounded transfer policy.

### PR 14 — end-to-end resident prefill/decode route

- `runtime_api=opencl`, exact selected A770, no CPU model-operation fallback.
- Full operation/state ownership receipt.

### PR 15 — quality, long-session, resource, and performance qualification

- Re-run accepted parity/quality/reliability/benchmark gates specifically for the full-resident route.

### PR 16 — full-residency promotion

- Update exact route/support/claim ledgers only after every required operation and state object has proof.

Each large subpath should be split into its own child issue before implementation; this umbrella defines the order and promotion boundary.

## Acceptance criteria

- Every required operation/state ID from #1903 has `observed_owner=a770` or an explicitly permitted non-model control-plane owner.
- No hidden CPU model-operation fallback, readback island, replacement map, output bias, diagnostic knob, or approximate semantics are used.
- Device tensors/state carry exact model, session, context, layout, and generation identity.
- Intermediate and final outputs satisfy CPU/upstream/generated-token contracts.
- KV cache positions, prompt isolation, reset, cancellation, and recovery are proven across short/long/repeated sessions.
- Host↔device transfers are limited to declared input/control/output boundaries and are receipt-visible.
- Peak/stable VRAM/RSS, allocations, kernel/transfer timing, power, and thermal behavior are qualified.
- Full-resident route has its own two-run same-device history and cannot inherit the hybrid benchmark.
- Support/claim docs are generated from accepted receipts and remain exact-profile only.

## Required evidence

- per-operation CPU/A770 parity fixtures;
- real official-model intermediate checkpoint packets;
- full operation/state ownership and transfer graph;
- generated-token/quality matrices;
- long decode/session/recovery receipts;
- current-head full-resident benchmark and history packet.

## Non-goals

- generic Intel/OpenCL HAL standardization;
- Arc 140V/NPU/OpenVINO/CUDA/Apple support;
- model-family breadth;
- production server/SLA;
- broad A770 or all-Intel residency claims.

## Claim boundary

Only this issue’s final accepted packet may promote **full-device-resident A770 inference**. Hybrid QK256 acceleration, resident weights, device activation quantization, or one resident attention subpath cannot independently make that claim.
